### PR TITLE
Update argo to 2.11.1

### PR DESCRIPTION
**Pull request by tumido**
Upgrades to Argo 2.11.1. This is a huge upgrade, the difference between 2.8.2 and 2.11.1 is night and day - much more polished product allowing simpler workflow definitions + more control over it. :slightly_smiling_face: 

Requires a new CRD on affected clusters - `WorkflowEventBinding` since Argo can now listen for events and webhooks (means there may be no need Argo Events anymore).

Only thing that's not working as expected, is submitting `CronWorkflow`s from UI, which is a permission issue - solved in upstream https://github.com/argoproj/argo/pull/4112 

^ just a minor inconvenience, will be probably fixed in later revision/patch release, there's no need to diverge from upstream just to apply a workaround for a barely used feature.

Metric exposing is fixed now, there's no need to hardcode it in the configmap.

### CODE REVIEW

**Comment by tumido** [comment]
cc @HumairAK

**Review by HumairAK — APPROVED** [review]

**Comment by anishasthana** [comment]
@tumido what is this waiting on?

**Comment by tumido** [comment]
I think I was waiting for PSI and for `WorkflowEventBinding` CRDs, but nah, it can work without them and we'll add those later as a part of migration to OCP4.

I'll update the release to 2.11.1 (released 3 days ago - adds some fixes for 2.11.0) and we can "unhold" this.

**Comment by tumido** [comment]
@anishasthana  done! :boom:

**Comment by tumido** [comment]
I'm gonna also open similar PR against upstream ODH, so we can use the upstream component instead of a custom Argo later on.

File: argo/base/workflow-controller-configmap.yaml [hunk 1]
```diff
@@ -4,7 +4,3 @@ metadata:
   name: workflow-controller-configmap
 data:
   containerRuntimeExecutor: k8sapi
-  metricsConfig: |
```
Comment (anishasthana) [inline argo/base/workflow-controller-configmap.yaml:7]: Any reason why we're disabling metrics?

**Pull request merged**